presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotConfig.java

```diff
@@ -38,6 +38,7 @@
     public static final int DEFAULT_MIN_CONNECTIONS_PER_SERVER = 10;
     public static final int DEFAULT_THREAD_POOL_SIZE = 30;
     public static final int DEFAULT_NON_AGGREGATE_LIMIT_FOR_BROKER_QUERIES = 25_000;
+    public static final int DEFAULT_LIMIT_THRESHOLD_FOR_TOP_BROKER_QUERIES = 1;
```

Comment:
shall we use a bigger number as default? e.g. 1000

Reply:
we had a previous discussion, this value could be varies based on user's query, so it would be better to disable it by default and enable it based on the query.

Reply:
Let me further clarify: Basically, Its not 100% safe to remove sorting from Presto and delegate that entirely to the connector. We are not sure of the root reason behind this. 

What happen is that on getting these many records (ie limitThreshholdForTopBrokerQueries), Presto might end up reordering them. And the absence of the sort makes this reordering wrong: There is no final sort at the end to correct the required order of the data.

What's more that this "limit" under which the presto might end up reordering can depend on the width of the data returned. So we can't be sure that it should be say 1000. The only thing safe here is to disable sort pushdown entirely . The way that is done here is to only push down if there is already a LIMIT 1 (making the sorting order moot).

Another way to implement the PR would have been to simply remove the TopN pushdown logic entirely and/or boolean config disable it.

Reply:
oh, get it. thank you for the explanation, @agrawaldevesh 
if I understand correctly, setting DEFAULT_LIMIT_THRESHOLD_FOR_TOP_BROKER_QUERIES > 1, might trigger incorrect result due to topN pushdown? please fix me if I mis-understand
I am inclined to have a config to disable topN pushdown, instead of relying on default value to guarantee correct behavior. what do you think @fx19880617 @agrawaldevesh

Reply:
I feel it's better to have the config, rather than disable it.
E.g. I want to query for `select sum(a) as sum_a from myTable order by sum_a desc limit 10`
then it's actually we can push down the entire query to pinot.

Reply:
One discussion we kept having when we were building this was: 

If the number of rows being returned by Pinot is small, then the sort in Presto has a very low cost and thus there is not much benefit in pushing down the sort. The benefit is only pronounced when we have a potentially high cardinality group by (not a global aggregation like above which produces only a single row), because its only there that we avoid fishing out the whole group-by from Pinot only to sort and select the top-K rows from it in Presto.

But unfortunately, we don't know what value of "K" is safe to be pushed down ? Is it 10, 100, 1000 ? How would our answer change with large widths etc. I wish we really understood the root cause here :-)

Reply:
yep, am inclined to have a config to control whether we do topN pushdown, default to be false.
For @fx19880617 's case, we could enable topN pushdown for the safe queries.
what do you think?

Reply:
right, I feel we should still keep the possibility for users to optimize the query if needed and possible

Reply:
ah, what you mean is to make this a session config and let user to decide turn it on/off instead of using a limit number ?

Reply:
yes, instead of using a configurable limit number, let's have a boolean session property, which enables topN pushdown, defaults to be false.
so that default will guarantee correctness. For advanced users who know their queries could benefit from pushdown, they could turn it on query by query.
exposing the limit number requires more domain knowledge for users, in addition to the risk of correctness

Reply:
done.